php/class-admin-menu.php

```diff
@@ -43,26 +43,62 @@ public function register_menu_items() {
 			[
 				'posts_per_page' => 99,
 				'post_type'      => hello_admin()->post_type->slug,
+				'meta_key'       => 'sub_menu',
+				'orderby'        => 'meta_value_num',
+				'order'          => 'ASC',
 			]
 		);
 
 		while ( $admin_pages_query->have_posts() ) {
 			$admin_pages_query->the_post();
-			$menu_position = get_post_meta(
+			$menu_position = (int) get_post_meta(
 				get_the_ID(),
 				hello_admin()->post_type->menu_position_key,
 				true
 			);
 
-			add_menu_page(
-				get_the_title(),
-				get_the_title(),
-				'read',
-				$this->prefix . get_post_field( 'post_name' ),
-				[ $this, 'render_menu_page' ],
-				'',
-				$menu_position
+			$sub_menu = get_post_meta(
+				get_the_ID(),
+				hello_admin()->post_type->sub_menu_key,
+				true
 			);
+
+			if ( '' === $sub_menu ) {
+				update_post_meta( get_the_ID(), hello_admin()->post_type->sub_menu_key, 0 );
+			}
+
+			$title     = get_the_title();
+			$menu_slug = $this->prefix . get_post_field( 'post_name' );
+
+			if ( ! $sub_menu ) {
```

Comment:
We could flip the logic here to make it a bit easier to read. Do the `if ( $sub_menu )` first, then follow with an `else`.